docs/developer-guide.rst

```diff
@@ -0,0 +1,116 @@
+.. _developer-guide:
+
+Developper guide
+----------------
+
+This page describes how to set up the development environment for working on
+c2cgeoform. It is for developers working on c2cgeoform itself, not for
+developers working on c2cgeoform-based applications.
+
+Note that c2cgeoform is a framework and with a
+`Pyramid  scaffold <https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/scaffolding.html>`_
+to create c2cgeoform-based applications. This scaffold produce a fully
+functional c2cgeoform-base project: the c2cgeoform_demo project.
+
+When running code checks and tests, these jobs are first run on the c2cgeoform
+framework itself where after the c2cgeoform_demo project is generated in `.build`
+folder and finally checks and tests are launched on this project too.
+
```

Comment:
When running code checks and tests, these jobs are first run on the c2cgeoform
framework itself. Then the c2cgeoform_demo project is generated in `.build`
folder. Finally, the checks and tests are also launched in this project.